### DISCUSSION TITLE
Can a code block access the above markdown content?

### DISCUSSION THREAD

**Discussion by coinyon** [quarto-dev/quarto-cli#6810] [Q&A]
### Description

I was wondering if there was any way for a code block to load the source document (or at least the sections above the current code block)? 

The use case would be that I want to enter some data in a human friendly format (e.g. a list, check box list, table, text) and then have some analysis in the following code blocks.

**Comment by mcanouil** [comment]
You can't because it would require several passes.
Why don't you output the text from a code cell? This way you can also "analyse" it.

**Reply by mcanouil** [reply]
You can include markdown file directly with `{{< include your-file.md >}}` and do in a R code cell (or Python) `readLines("your-file.me")`.
There are many ways to read/output external file which are really related to Quarto itself.

What you can't do is for a document to read itself or more accurately you should not do that.

**Reply by mcanouil** [reply]
To be clear, it's simply that doing this kind of recursive things with files is usually not handled well (for obvious reasons) by several file systems, but if you have an external file, you can easily include and read it in a code cell.

**Reply by coinyon** [reply]
Sorry, used wrong account for posting. Here are my responses for reference.

--

Thanks for your response.

I'm not sure if giving access to the source document really would require several passes. The easiest solution would be to have access to the source .md file within the script (similar to `__file__` in Python). Then I could open it and read it again.

My use case is that I already manage some data (todo list, expenses, ...) in markdown files. It would be neat if I could add a code block below that to show for example expenses total or so. But I agree that this is probably not the intended use case for Quarto.

--

Ok thanks. Will probably try to get it working this way anyway for fun, but I'll remember that it is unsupported/discouraged :)

**Reply by cderv** [reply]
> My use case is that I already manage some data (todo list, expenses, ...) in markdown files. It would be neat if I could add a code block below that to show for example expenses total or so. But I agree that this is probably not the intended use case for Quarto.

You could maybe use Lua filter to access some content of block in your document and process it you way to need to remove from output
and add your formatted result. 

Though you would need to do your processing from Lua and not a computation engine ran by Quarto). 

Anyhow, I don't think this is something that Quarto will cover - you could try build something using Lua extension system, or though computation maybe (**knitr** would allow to build custom engine that could be retreived and parsed from R to be used in a subsequent chunk).

**Reply by kcarnold** [reply]
I think the original suggestion was to do something like this:

````
```{python}
#| output: asis
my_data = r'''
some data
'''
print(my_data)
```

```{python}
analyze(my_data)
```
````

You'd just have to accept editing the markdown within that raw triple-quoted string
